test/helpers/data_sets_helper_test.rb

```diff
@@ -0,0 +1,171 @@
+
+    resdata = [{ '106' => 'A' }, { '106' => 'B' }, { '106' => 'C' }]
+    assert_similar_arrays data, resdata
+  end
+
+  test 'slickgrid data population w/ data' do
+    project = Project.find_by_name 'slickgrid_project_no_latlon'
+    dataset = DataSet.find_by_name 'Slickgrid Dataset 2'
+    cols, data = format_slickgrid_merge project.fields, dataset.data
+    cols_pop, data_pop = format_slickgrid_populate cols, data
+
+    assert_similar_arrays cols, cols_pop
+    assert_similar_arrays data, data_pop
+  end
+
+  test 'slickgrid data population w/o data' do
+    project = Project.find_by_name 'slickgrid_project_no_latlon'
+    dataset = DataSet.find_by_name 'Slickgrid Dataset 3'
+    cols, data = format_slickgrid_merge project.fields, dataset.data
+    cols_pop, data = format_slickgrid_populate cols, data
+    data_pop = [{ id: 0, '106' => '' }]
+
+    assert_similar_arrays cols, cols_pop
+    assert_similar_arrays data, data_pop
+  end
+
+  test 'slickgrid editor association' do
+    project = Project.find_by_name 'slickgrid_project'
+    dataset = DataSet.find_by_name 'Slickgrid Dataset 1'
+    cols, data = format_slickgrid_merge project.fields, dataset.data
+    cols, data = format_slickgrid_populate cols, data
+    cols, data = format_slickgrid_editors cols, data
+
+    rescols = [
+      { id: '"slickgrid-100"', name: '"text_field"', field: '"100"', editor: 'TextEditor', restrictions: '""', sortable: 'false' },
+      { id: '"slickgrid-101"', name: '"restrictions_field"', field: '"101"', editor: 'TextEditor', restrictions: '["A", "B", "C"]', sortable: 'false' },
+      { id: '"slickgrid-102"', name: '"number_field"', field: '"102"', editor: 'NumberEditor', restrictions: '""', sortable: 'false' },
+      { id: '"slickgrid-103"', name: '"timestamp_field"', field: '"103"', editor: 'TimestampEditor', restrictions: '""', sortable: 'false' },
+      { id: '"slickgrid-104-105"', name: '"latitude_field & longitude_field"', field: '"104-105"', editor: 'LocationEditor', restrictions: '""', sortable: 'false' }
+    ]
+    assert_similar_arrays cols, rescols
+
+    resdata = '[{"100":"A","101":"A","102":"1","103":"11","104-105":"0, 0"},' \
+      '{"100":"B","101":"B","102":"2","103":"22","104-105":"1, 1"},' \
+      '{"100":"C","101":"C","102":"3","103":"33","104-105":"2, 2"}]'
+    assert data == resdata
+  end
+
+  test 'slickgrid editor association oops' do
+    project = Project.find_by_name 'slickgrid_oops'
+    dataset = DataSet.find_by_name 'Slickgrid Dataset Ooops I Messed Up'
+    cols, data = format_slickgrid_merge project.fields, dataset.data
+    cols, data = format_slickgrid_populate cols, data
+    cols, data = format_slickgrid_editors cols, data
+
+    rescols = [
+      { id: '"slickgrid-107"', name: '"oops"', field: '"107"', editor: 'Slick.Editors.Text', restrictions: '""', sortable: 'false' }
+    ]
+    assert_similar_arrays cols, rescols
+
+    resdata = '[{"107":"SOMEBODY ONCE TOLD ME THE WORLD IS GONNA ROLL ME"}]'
```

Comment:
Is there a way to get this string from the original file that declares it without having to hardcode it here too?